autobahn/twisted/wamp.py

```diff
@@ -84,7 +85,29 @@ def _reject_future(future, error):
 
     @staticmethod
     def _add_future_callbacks(future, callback, errback):
-        return future.addCallbacks(callback, errback)
+        # callback and/or errback may be None
+
+        # Note that errback takes 3 args: type, exception, tb our
+        # errbacks can be consistent between Twisted and asyncio.
```

Comment:
Looking at the code in Twisted's Failure, for example, my opinion would be that we should provide at least the 3 args from sys.exc_info() -- otherwise, I don't know of any way that you could print out a stack trace in your error handler.

All you'd be able to do is print out the Exception instance.

As I noted, another alternative would be to provide our own Failure-type object for asyncio (and then in Twisted you could get your Failure and in asyncio you'd get "some object that has many of the same methods"). Not sure if this is a "win" or not though ;)